File: server/src/main/java/moe/yuuta/server/MainVerticle.java
package moe.yuuta.server;

import java.util.Arrays;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.CompositeFuture;
import io.vertx.core.DeploymentOptions;
import io.vertx.core.Future;
import moe.yuuta.server.api.ApiVerticle;
import moe.yuuta.server.topic.TopicRegistry;

@SuppressWarnings("unused")
public class MainVerticle extends AbstractVerticle {
    @Override
    public void start(Future<Void> startFuture) {
        DeploymentOptions options = new DeploymentOptions().setConfig(config());
        CompositeFuture.all(Arrays.asList(
                Future.<CompositeFuture>future(f -> TopicRegistry.getInstance().init(vertx, f)),
                Future.<String>future(f -> vertx.deployVerticle(ApiVerticle::new, options, f))
        )).setHandler(ar -> {
            if (ar.succeeded()) startFuture.complete();
            else startFuture.fail(ar.cause());
        });
    }
}

File: server/src/main/kotlin/moe/yuuta/server/MainVerticle.kt
package moe.yuuta.server

import io.vertx.core.*
import moe.yuuta.server.api.ApiVerticle
import moe.yuuta.server.topic.TopicRegistry
import java.util.*
import java.util.function.Supplier

/**
 * Automated converted to Kotlin by Android Studio on Jan. 2 / 2019, not verified.
 */
class MainVerticle : AbstractVerticle() {
    override fun start(startFuture: Future<Void>) {
        val options = DeploymentOptions().setConfig(config())
        CompositeFuture.all(Arrays.asList(
                Future.future<CompositeFuture> { f -> TopicRegistry.get().init(vertx, f) },
                Future.future<String> { f -> vertx.deployVerticle(Supplier<Verticle> { ApiVerticle() }, options, f) }
        )).setHandler { ar ->
            if (ar.succeeded())
                startFuture.complete()
            else
                startFuture.fail(ar.cause())
        }
    }
}
